### ISSUE TITLE
[protocol/exporter]: should `user-agent` header be configurable via `OTEL_EXPORTER_OTLP_HEADERS`?

### ISSUE THREAD

**Issue by david-luna** [open-telemetry/opentelemetry-specification#4230]
We had this question in yesterday's JS SIG (2024-09-25) and originates from the issue I've opened a while ago https://github.com/open-telemetry/opentelemetry-js/issues/4447

From [the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/d50fcf5d37ea06a47017a3fe60b47a407fffb669/specification/protocol/exporter.md?plain=1#L50):
> Key-value pairs to be used as headers associated with gRPC or HTTP requests.

`User-Agent` is a header too but in some implementations exporters usually set it up following the [recommendation](https://github.com/open-telemetry/opentelemetry-specification/blob/d50fcf5d37ea06a47017a3fe60b47a407fffb669/specification/protocol/exporter.md?plain=1#L191) in the spec and it cannot be modified see:
- https://github.com/open-telemetry/opentelemetry-dotnet/blob/9cca0ac8f1e5ca5b4d17494b7120d443f997fdf2/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs#L31
- https://github.com/open-telemetry/opentelemetry-js/blob/2a4919c1cf99d3403d387d7589836fd9e3018896/experimental/packages/opentelemetry-exporter-metrics-otlp-proto/src/OTLPMetricExporter.ts#L37

Should we allow to override the header although the user might not follow the spec?

**Comment by dmathieu** [comment]
[RFC 7231](https://www.rfc-editor.org/rfc/rfc7231#page-46) states:

> The User-Agent field-value consists of one or more product identifiers

In the case of OTLP exporters, the product identifier is OpenTelemetry, not the user's application, despite it happening within the said app.
So even if some folks may want to override it, and maybe we want to support it, the formal recommendation should probably not be to do so.

**Comment by david-luna** [comment]
Thanks @dmathieu and I agree with you.

I guess the only use case for having access to user agent is if you're creating your own product (like a distribution) based on the SDK an you want to add that info in it right?

**Comment by dmathieu** [comment]
Do we need it configurable via an environment variable then?
A distribution could modify a local variable, or any other mean to change the value that wouldn't easily be available to end-users.

**Comment by david-luna** [comment]
> Do we need it configurable via an environment variable then?
> A distribution could modify a local variable, or any other mean to change the value that wouldn't easily be available to end-users.

Yes, a local variable would be enough. Thanks for your feedback :)
